### ISSUE TITLE
Deployment fails and Functions return "The service is unavailable"

### ISSUE THREAD

**Issue by maersk96** [Azure/static-web-apps#1097]
**Describe the bug**
Starting this week all my production Azure Static Web App services have been returning "The service is unavailable" responses. I am also experiencing issues with deployment to these SWA services. This is the DevOps deployment response:

2023-03-08T09:46:48.8086600Z ##[section]Finished Upload. Polling on deployment.
2023-03-08T09:46:48.8556874Z Status: InProgress. Time: 0.0467032(s)
2023-03-08T09:47:03.9068232Z Status: InProgress. Time: 15.0980397(s)
2023-03-08T09:47:18.9516267Z Status: InProgress. Time: 30.1429502(s)
2023-03-08T09:47:34.0105878Z Status: Failed. Time: 45.2015801(s)
2023-03-08T09:47:34.0106351Z ##[error]Deployment Failed :(
2023-03-08T09:47:34.0106747Z ##[error]Deployment Failure Reason: Failed to deploy the Azure Functions.

I cannot find any official status message stating that the SWA service is down, so I am suspecting that the issue lies within our setup/subscription. Is anyone experiencing the same issue? Did any breaking changes happen in regards to SWA services that i am unaware of?

**To Reproduce**
1. Create and deploy an SWA with Functions
2. Check if SWA deployment fails
3. If not, check if Functions return "Service unavailable" response

**Comment by maersk96** [comment]
> Also facing the same issue - doubled by the fact that for some existing SWA's the functions are not running.

Exactly my issue. My Static Web App frontends are still reachable to some extend, but Functions are unavailable.

I am wondering why nothing shows up on the [Azure status page](https://azure.status.microsoft/en-us/status)

**Comment by KongMats** [comment]
Also having this issue at the moment. Getting the error "Deployment Failure Reason: Failed to deploy the Azure Functions."

**Comment by polaroi8d** [comment]
> Also having this issue at the moment. Getting the error "Deployment Failure Reason: Failed to deploy the Azure Functions."

Same issue.

**Comment by valstu** [comment]
Facing the same issue

**Comment by myticketgit** [comment]
We have the same issue :-(

**Comment by erikknorren** [comment]
We have the same issue. Multiple tenants, free plan/standard plan/enterprise grade edge. Sent support ticket via Azure.

**Comment by ottosulin** [comment]
We have been experiencing a total outage of Static Web Apps at least since 08:15 UTC this morning. I'm honestly very surprised to see that Azure has not yet communicated about this to customers with service being totally out for at least 2h30min now and counting.

**Comment by matthiasbruch** [comment]
Deployment Failed :(
Deployment Failure Reason: Failed to deploy the Azure Functions.

For further information, please visit the Azure Static Web Apps documentation at https://docs.microsoft.com/en-us/azure/static-web-apps/
If you believe this behavior is unexpected, please raise a GitHub issue at https://github.com/azure/static-web-apps/issues/
Exiting
##[debug]Exit code 1 received from tool '/usr/bin/bash'
##[debug]STDIO streams have closed for tool '/usr/bin/bash'
##[debug]task result: Failed
##[error]Error: The process '/usr/bin/bash' failed with exit code 1
##[debug]Processed: ##vso[task.issue type=error;]Error: The process '/usr/bin/bash' failed with exit code 1
##[debug]Processed: ##vso[task.complete result=Failed;]Error: The process '/usr/bin/bash' failed with exit code 1
Finishing: AzureStaticWebApp

**Comment by erikknorren** [comment]
@ChristianFahrner we have had a similar issue on 14 december 2022, with the same consequences.

**Comment by erijss** [comment]
@erikknorren: Did you found a solution or workaround that you can share?

**Comment by eodeg** [comment]
Experiencing the sam issue

**Comment by hohenp** [comment]
Same issue here. The site stopped working today. Only message "The service is unavailable." is shown.
Redeployment fails with message "Deployement failed :("
Anything we can do here?

**Comment by andrem0** [comment]
Same here. Running Next.js.

**Comment by erikknorren** [comment]
@erijss No it seemed back then as though it was downtime.

**Comment by matthiasbruch** [comment]
Are you guys with nextjs also using this preview feature?

**Comment by Even-MW** [comment]
Same issue here, Nextjs 13 with Static Web App (preview)

**Comment by ottosulin** [comment]
Got hold of Azure Support on Twitter, they confirmed they're looking into this.

**Comment by tlaundal** [comment]
Same issue here. Static assets work, but any function invocation yields a 503 response with the text "The service is unavailable.".

**Comment by hohenp** [comment]
Our page is back running since some minutes. Anyone knows what happened?

**Comment by vd-aloker** [comment]
Same issue here. Deployment fails. Existing deployments that use functions see outages (503 The service is unavailable).

**Comment by cfillies** [comment]
I am having the same issue. no API and no deployment

**Comment by sshehab1** [comment]
same here

**Comment by filipkowal** [comment]
Same issue. Next.js 13 with appDir

**Comment by erikknorren** [comment]
Hello Erik,

From case verbatim we understand that you are facing issue while accessing Azure App Services.

I would like to inform you that seems like an issue going on in West Europe region and our product group team is actively working towards the fix and we are expecting an update soon. 

Impact Statement: Starting at 07:18 UTC on 08 Mar 2023, you have been identified as a customer using App Service in West Europe who may receive intermittent HTTP 500-level response codes, experience timeouts or high latency when creating or deleting containers for resources hosted in this region. Existing containers would not be impacted by this issue.

Current Status: Our investigation indicates that some container instances which App Service utilizes became unresponsive due to an internal backend service becoming unhealthy. Impact is limited to resources hosted in the West Europe region. This caused operational functions to stop processing and resulted in App Service users experiencing the errors mentioned above. 

We have performed recovery actions to help the internal backend service return to a healthy state. We are currenting monitoring the error rates to validate its efficiency. In parallel, we're exploring alternative options to clean up the backlog of stuck requests. 

We are constantly following up with them and shall get back to you as soon as we have further update to share with you.

For latest official updates on this outage please visit: https://azure.status.microsoft/th-th/status/

Apologies for the inconvenience caused.

**Comment by costinBol** [comment]
> 

Yeah, that's really frustrating for me, the [Azure status ](https://azure.status.microsoft/status) page doesn't seem to detect any incidents. Not sure how they managed to not detect this, since it looks like the complete Static Web App service is down, you can just ping any website and get the error "The service is unavailable."

**Comment by niicojs** [comment]
Same here. Not nextjs but a Javascript function so it may be related.

**Comment by craftzneko** [comment]
Same here, nextjs, Deployment Failure Reason: Failed to deploy the Azure Functions

**Comment by louison77** [comment]
Same here.

Deployment Failed :(
Deployment Failure Reason: Failed to deploy the Azure Functions.

**Comment by craftzneko** [comment]
I am now able to build, anyone else working again?

**Comment by hohenp** [comment]
Yes, our two static sites are back online for the moment. But they have been online this afternoon for a couple of minutes. We are waiting.. :)

**Comment by mbalslow** [comment]
Same issue

**Comment by costinBol** [comment]
Both West Europe and UK South sites are back online now for us.

**Comment by kjellkarlsson** [comment]
West Europe works for me

**Comment by pstovik** [comment]
works for me as well
this was MS Support response
> Thank you for contacting Microsoft Support. I am *** from the Azure App Service team; I would be the support Engineer that would assist you with this service request. I understand you are currently experiencing 503 error with service unavailable on your application.
Our investigation suggests a backend service that App Service utilizes became unhealthy which resulted in container creation errors for App Service customers in the West Europe region. This caused operational functions to stop processing and resulted in users experiencing the errors.
We have performed recovery actions to the backend service to return to a healthy state. We are currenting monitoring the error rates to validate the progress.
Our sincere apologies for the inconvenience this may have caused you at this time.
> Kindly reach out to me if you notice any change from your end. However, I will continue to monitor the situation from our end and keep you updated.
> I apologize once again for the inconvenience this may have caused.

**Comment by artlepool** [comment]
working for us now also

**Comment by myticketgit** [comment]
The problem is back - I am starting to have serious trust issues with static websites! Any updates?

**Comment by costinBol** [comment]
> 

I'm having the issue with West Europe. UK South still works fine.

**Comment by ksdaniel** [comment]
I can also confirm - all my SWA's are down in West Europe.

**Comment by hohenp** [comment]
Our sites were down this morning. In the moment they are back running. So, SWA cant be used for production!

**Comment by myticketgit** [comment]
> Our sites were down this morning. In the moment they are back running. So, SWA cant be used for production!

Yeah - I am coming to the same conclusion - we will definitely move to something else. Anyone knows if it was "just" EU-WEST again or it was all SWA's across regions?

**Comment by hohenp** [comment]
And down again... all SWA. Any news or updates??

**Comment by matthiasbruch** [comment]
Ours is still up

Gesendet von Outlook für iOS<https://aka.ms/o0ukef>
________________________________
Von: hohenp ***@***.***>
Gesendet: Wednesday, March 15, 2023 12:34:27 PM
An: Azure/static-web-apps ***@***.***>
Cc: Matthias Bruch ***@***.***>; Comment ***@***.***>
Betreff: Re: [Azure/static-web-apps] Deployment fails and Functions return "The service is unavailable" (Issue #1097)


And down again... all SWA. Any news or updates??

**Comment by ksdaniel** [comment]
We are also affected with all 5 apps.

 Any idea how we can escalate this?

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by hohenp** [comment]
@Reshmi-Sriram Do you have infos regarding the current breakdown of the SWA again?

**Comment by annikel** [comment]
Hi @hohenp, @ksdaniel, @matthiasbruch, @myticketgit and @costinBol 

The issue with deploying to Azure Function Apps in West Europe, has now been resolved. We apologize for any inconvenience this issue may have caused. 

If you are still experiencing any problems, please don't hesitate to reach out. We're here to help unblock you and address any ongoing concerns.

**Comment by costinBol** [comment]
> 

@annikel - I guess there are two concerns here:
1. That the service keeps going down for whole regions, so people lose confidence in the 99.9% uptime promise
2. That the Azure Status website looks hard-coded on green (or, as @ksdaniel put it, it looks "static"), it's rarely informative - and this lack of communication and transparency further lowers confidence.

**Comment by morganfeeney** [comment]
It’s down again in the EU region. Gave this a shot but will use another service for natraj due to the unreliability.

**Comment by tlaundal** [comment]
> It’s down again in the EU region. Gave this a shot but will use another service due to the unreliability.

Same here. 503 Service Unavailable for our node API routes. Static part of site is still up. Have not tested deployments.

Submitted a support request to Azure.

**Comment by chribjel** [comment]
Down again currently...

**Comment by ksdaniel** [comment]
I can confirm. We are down as well. This is freaking unbelievable.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by cfillies** [comment]
My Customers are getting more and more annoyed

**Comment by DavidAtImpactCubed** [comment]
More outages for us today as well, affecting our production environment Linux consumption plan function app in West Europe. Other regions (including North Europe) didn't seem affected last week, when we migrated one faulting function app to NE.

**Comment by espenja** [comment]
> Hi @hohenp, @ksdaniel, @matthiasbruch, @myticketgit and @costinBol
> 
> The issue with deploying to Azure Function Apps in West Europe, has now been resolved. We apologize for any inconvenience this issue may have caused.
> 
> If you are still experiencing any problems, please don't hesitate to reach out. We're here to help unblock you and address any ongoing concerns.

My biggest issue with all of this is that the Azure Status page stays green ✅ even when you seemingly know you are having issues. What is the point of the status page if it doesn't reflect the current status of ops? If we can immediately see that Azure is having issues, then we can tell our customers that the problems are most likely not our fault and that it will resolve itself when Microsoft fixes whatever is going on. The lack of transparency about downtime is very frustrating. Some time ago our Azure Functions were down and we were trouble shooting for a good hour before we decided to create an A level ticket. When creating the ticket we weren't allowed to actually submit the ticket because "There is an active ongoing investigation with the resource type selected" (paraphrasing), yet the status page was all green and shiny.

Last week's huge outage in West-Europe didn't get any status updates or warnings until the issue was already well on the way to being resolved.

**Comment by kabforks** [comment]
> Our investigation suggests a backend service that App Service utilizes became unhealthy which resulted in container creation errors for App Service customers in the West Europe region.

Perhaps you should update your Azure Status Page when all customers in West Europe region experience errors? If not, I would be most interested to hear what actually qualifies as "Warning" or "Critical". I have no confidence in the Azure Status page going forward. Empty shell.

**Comment by Erlpil** [comment]
We are now down again... This is crazy.

**Comment by fwmoor** [comment]
Same issue again... It's getting ridiculous now

**Comment by vd-aloker** [comment]
That's really unfortunate. Sadly, we have lost our confidence (for the time being) in SWA as a platform for production work loads.

**Comment by costinBol** [comment]
> Same issue again... It's getting ridiculous now

Same here - both UK South and West Europe getting "The service is unavailable.".
The Azure Status page reassuringly shows all the app services as green as before. There is one warning though:
Information
Azure Resource Manager - Azure Resource Manager Operatons failures - Investigating

**Comment by ksdaniel** [comment]
I can also confirm - our sites are down.

**Comment by ottosulin** [comment]
Here we go again. Azure should now clearly communicate that they have identified the root causes and are doing something about them, if they want customers to retain confidence in the service.

**Comment by maersk96** [comment]
Aaaand, we're down again 🎉🎉🎉

**Comment by tlaundal** [comment]
Same here

**Comment by ksdaniel** [comment]
+1

**Comment by erikknorren** [comment]
Down again as well.

**Comment by erikknorren** [comment]
https://twitter.com/AzureSupport/status/1653696849583169540 Azure support replied to me

**Comment by thomas-bingel** [comment]
+1

**Comment by olarsson** [comment]
this is not good, this is a major outage and its lasting a very long time. this effectively kills any production-environment. why does this keep occuring? i was considering moving our future web infrastructure to static web apps, but im very hesitant now.

**Comment by nanonandi** [comment]
We are just about to ship to production and this happens... idk what to think :S

**Comment by ottosulin** [comment]
Our SWAs are down once again, like they were also yesterday.

Microsoft should begin to at least acknowledge in the Service Health when the service is not working. That would be a start and the bare minimum Microsoft should be doing. If these severe outages (without any communications!) continue we will have to consider alternatives.

**Comment by ksdaniel** [comment]
This feels more and more like a social experiment: "how low can you push service levels before customers start leaving". 

We feel very betrayed...

**Comment by hohenp** [comment]
yes.. down again... Hello Microsoft... Is there anyone not working on the CoPilot stuff and cares about the other services around? A info about the reasons and status would be the least to inform about...

**Comment by fwmoor** [comment]
We've moved away from SWA now. It just not stable enough

**Comment by costinBol** [comment]
Just out of curiosity, which other providers have you all considered? I was looking at Hostinger and Vercel since they work well with SvelteKit but I'd like to keep my current Azure DevOps CI/CD pipelines,

**Comment by projectsbydan** [comment]
down in germany as well. paid plan. next.js swa.
a redeploy seems to "wake it up".

**Comment by niels-ws** [comment]
Down for us. Deployment fails with "Deployment Failure Reason: Failed to deploy the Azure Functions."

**Comment by TheSirLan** [comment]
Re-running the deployment job worked for me just then - I am up again.

This has been intermittent for a couple of days now though, I will open my app and see "The service is unavailable."

Re-deploying sometimes brings it back up right away or it just fails to deploy.

While it's in this state I notice that the APIs tab in the SWA resource shows no backend function assigned at all.

Hopefully these stability issues can be fixed soon.

**Comment by TheSirLan** [comment]
> Re-running the deployment job worked for me just then - I am up again.
> 
> This has been intermittent for a couple of days now though, I will open my app and see "The service is unavailable."
> 
> Re-deploying sometimes brings it back up right away or it just fails to deploy.
> 
> While it's in this state I notice that the APIs tab in the SWA resource shows no backend function assigned at all.
> 
> Hopefully these stability issues can be fixed soon.

5 minutes later, down again

**Comment by costinBol** [comment]
Has this started again? We've been getting "The service is unavailable." for at least 3h now. Wondering if it's only UK South/West or other regions as well.

**Comment by cfillies** [comment]
Same 503 in "West Europe" as soon as  it tries to access app

**Comment by ottosulin** [comment]
It's ridiculous how Microsoft engineers can ignore these here.

I suggest everyone keeps making support tickets and remember to ask for refund for breaking the SLA. Because I guess that is the only way of getting attention of some director in Redmond guarding his/her bonus who thinks fixing these issues is not important as integrating Copilot to yet another service.

**Comment by jikuja** [comment]
At least this time they created service health alerts without contacting support.

**Comment by peterboba** [comment]
We're also in contact with MS support since we're experiencing HTTP 503/502 responses when trying to access
https://func.scm.azurewebsites.net/. Mostly WestEurope, but seen also in other regions.

**Comment by Chizaruu** [comment]
Aye, same issue again.
![Image](https://github.com/user-attachments/assets/7766981f-4dc8-4a4e-8436-e85dff81967b)
![Image](https://github.com/user-attachments/assets/3029744f-6526-4c17-ac85-e93615f1199d)
